Apps/Src/ReadCarCAN.c

```diff
@@ -189,53 +147,75 @@ void Task_ReadCarCAN(void *p_arg)
     );
     assertOSError(OS_READ_CAN_LOC, err);
 
-    //Start CAN Watchdog timer
+    // Start CAN Watchdog timer
     OSTmrStart(&canWatchTimer, &err);
     assertOSError(OS_READ_CAN_LOC, err);
-    
 
-    while (1)
-    {
+    #ifndef __TEST_READCARCAN
+    Minion_Error_t Merr; // Not used when testing since ignition won't be changed
+    #endif
+
+    while(1){
+
+        #ifndef __TEST_READCARCAN
+        arrayIgnitionStatus = Minion_Read_Pin(IGN_1, &Merr); // Don't check ignition if just testing
+        #endif
+
+        // Array Contactor is turned off if Ignition 1 is off
+        if(arrayIgnitionStatus == false){
+            Contactors_Set(ARRAY_CONTACTOR, OFF, true);
+
+            // Update Display
+            UpdateDisplay_SetArray(false);
+        }
```

Comment:
As said above, this logic (along with the "prechargeComplete" logic) can all be put in a helper.